test/integration/binary/launchers.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "launchers.hpp"
+
+#include <gtest/gtest.h>
+#include "common/byteutils.hpp"
+
+#include "bindings/model_crypto.hpp"
+
+using namespace boost::process;
+
+namespace binary_test {
+
+  constexpr auto cTimeToKill = std::chrono::minutes(15);
+
+  void Launcher::operator()(const std::string &example) {
+    ipstream pipe;
+    const auto &command = launchCommand(example);
+    if (command.empty()) {
+      FAIL() << "Launcher provided empty command";
+    }
+    child c(command, std_out > pipe);
+    auto terminated = c.wait_for(cTimeToKill);
+    if (not terminated) {
+      c.terminate();
+      FAIL() << "Child process was terminated because execution time limit "
```

Comment:
Is it always a test fail?

Reply:
Yes. The hang of external generator is considered as test case fail.